config/initializers/doorkeeper.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+Doorkeeper.configure do
+  orm :active_record
+  base_controller "ActionController::Base"
+
+  # Enabled grant flows
+  grant_flows %w[client_credentials]
```

Comment:
Usually client credentials grants aren't that often used, and instead authorization code is. Acting on behalf of the application and not the user is generally a bit weird, since it can lead to misauthorization or misattribution.

Reply:
Also, if you do support authorization code, be sure to use pkce:
```
pkce_code_challenge_methods ['S256']
```

(requires additional database stuff, see doorkeeper docs)

Reply:
Yeah, I was starting with client credentials because it's mainly intended for non-interactive machine-to-machine access to the app-owning user's stuff, so far at least. Happy to be told I'm doing it wrong though, parsing all the choices in OAuth is complex; I'll have another read into Authorization Code and see how I can do non-interactive client apps with it.

Reply:
yeah, for M2M client credentials is what you'd want, but usually many use-cases you want authorization code. Just be sure that when you're asserting scopes and fetching the user from the authorization, you're correctly handling whether you've client credentials or a regular access token.